apps/web/src/components/PoolCandidatesTable/SkillMatchDialog.tsx

```diff
@@ -0,0 +1,138 @@
+
+interface SkillMatchDialogProps {
+  filteredSkills: Skill[];
+  experiences: Experience[];
+  skillsCount: Maybe<number>;
+  poolCandidateName: string;
+}
+
+const SkillMatchDialog = ({
+  filteredSkills,
+  experiences,
+  skillsCount,
+  poolCandidateName,
+}: SkillMatchDialogProps) => {
+  const intl = useIntl();
+  const filteredSkillsTotal = filteredSkills.length;
+
+  if (filteredSkills.length === 0)
+    return (
+      <p
+        aria-label={intl.formatMessage(
+          {
+            defaultMessage: "{poolCandidateName} has 0 of 0 skills",
+            id: "4SNVtn",
+            description:
+              "Aria-label for the title displayed on the candidate skill count column.",
+          },
+          { poolCandidateName },
+        )}
+      >
+        {intl.formatMessage({
+          defaultMessage: "0 of 0",
+          id: "gkLEbN",
+          description:
+            "Title displayed on the candidate skill count column when no skills are selected.",
+        })}
+      </p>
+    );
+
+  return (
+    <Dialog.Root>
+      <Dialog.Trigger>
+        <Button
+          color="black"
+          mode="inline"
+          aria-label={intl.formatMessage(
+            {
+              defaultMessage:
+                "{poolCandidateName} has {skillsCount} of {filteredSkillsTotal} skills",
+              id: "4QEl+4",
+              description:
+                "Aria-label for the title displayed on the candidate skill count column.",
+            },
+            {
+              poolCandidateName,
+              skillsCount,
+              filteredSkillsTotal,
+            },
+          )}
+        >
+          {intl.formatMessage(
+            {
+              defaultMessage: "{skillsCount} of {filteredSkillsTotal}",
+              id: "WFhX9m",
+              description:
+                "Title displayed on the candidate skill count column.",
+            },
+            { skillsCount, filteredSkillsTotal },
+          )}
+        </Button>
+      </Dialog.Trigger>
+      <Dialog.Content>
+        <Dialog.Header>
+          {intl.formatMessage({
+            defaultMessage: "Skills matched in detail",
+            id: "qQZQj+",
+            description:
+              "Heading displayed on the candidate skill match dialog.",
+          })}
+        </Dialog.Header>
+        <Dialog.Body>
+          {filteredSkills.map((skill) => (
+            <SkillTree
+              key={skill.id}
+              skill={skill}
+              experiences={experiences}
+              showDisclaimer
+              disclaimerMessage={intl.formatMessage({
+                defaultMessage:
+                  "There are no experiences attached to this skill",
+                id: "pJqoQF",
+                description:
+                  "Disclaimer displayed in the skill tree on the skill match dialog.",
+              })}
+              hideConnectButton
+              hideEdit
+            />
```

Comment:
sorry, totally missed this on the first pass. Can we add `headingAs="h3"` right now we jump from `h2` in the title to `h4` in the tree 😬

Reply:
Nice catch! Fixed by [c32a80b](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7015/commits/c32a80b205504abe8258c4899130e17ae92fc41d)